libraries/chain/database_proposal_object.cpp

```diff
@@ -21,6 +21,18 @@ namespace golos { namespace chain {
         return find<proposal_object, by_account>(std::make_tuple(author, title));
     }
 
+// Yet another temporary PoC
+// TODO: This can be generalized to generate all 3: get_, find_ and throw_if_exists_
+// methods with variable number of params (like DEFINE_/DECLARE_API + PLUGIN_API_VALIDATE_ARGS)
+#define DB_DEFINE_THROW_IF_EXIST(O, T1, N1, T2, N2) \
```

Comment:
In some case ID for object consist from one name only. I'm not against this macro if these cases are also covered. However, I would try to replace these methods with the template function by adding the methods of the `make_<...>_id` for the used objects. However, this should not be done now due to time constraints.